### PR TITLE
Cargo: upgrade kvproto to release-3.0

### PR DESCRIPTION
**Pull request by overvenus**
## What have you changed? (mandatory)

Upgrade kvproto to release-3.0

## What are the type of the changes? (mandatory)

- Misc (other changes)

## How has this PR been tested? (mandatory)

Unit tests.

## Does this PR affect documentation (docs) or release note? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -122,7 +122,7 @@ cop_codegen = { path = "components/cop_codegen" }
 tipb_helper = { path = "components/tipb_helper" }
 panic_hook = { path = "components/panic_hook" }
 tipb = { git = "https://github.com/pingcap/tipb.git" }
-kvproto = { git = "https://github.com/pingcap/kvproto.git" }
+kvproto = { git = "https://github.com/pingcap/kvproto.git", branch = "release-3.0" }
```
Comment (siddontang) [inline Cargo.toml]: em, I think it is weird to depend on a release branch here. Master should depend on kvproto master.

Comment (overvenus) [reply Cargo.toml]: We will do a fast-forward merge a few days later and after that I will update to master.

Comment (siddontang) [reply Cargo.toml]: can you add a comment here?

**Pull request merged**